src/pkg/podman/podman.go

```diff
@@ -332,3 +333,83 @@ func SystemMigrate(ociRuntimeRequired string) error {
 
 	return nil
 }
+
+// internalError serves for representing errors printed by Podman to stderr
+type internalError struct {
+	errors []string
+}
+
+func (e *internalError) Error() string {
+	if e.errors == nil || len(e.errors) == 0 {
+		return ""
+	}
+
+	var builder strings.Builder
+	for i, part := range e.errors {
+		if i != 0 {
+			builder.WriteString(": ")
+		}
+		builder.WriteString(part)
+	}
+	return builder.String()
+}
+
+// Is lexically compares errors
+//
+// The comparison is done for every part in the error chain not across.
+func (e *internalError) Is(target error) bool {
+	if target == nil {
+		return false
+	}
+
+	if e.errors == nil || len(e.errors) == 0 {
+		return false
+	}
+
+	for {
+		if e.errors[0] == target.Error() {
+			return true
+		}
+
+		if e = e.Unwrap(); e == nil {
```

Comment:
Can't this be implemented as a `for ... range ...` loop? That could help with the signature of `Unwrap`.

Reply:
I wanted to follow the original implementation of `Is()`. Using a `for ... range ...` loop would work, too.

Reply:
One of the main advantages of implementing `internalError` as a slice of error strings is that the `Is` implementation, where we are matching against each element in the slice, becomes very straightforward. So it's good to take advantage of that.